spec/models/manageiq/providers/redhat/infra_manager/refresher_target_vm_spec.rb

```diff
@@ -0,0 +1,352 @@
+                              :filename => "da123bb9-095a-4933-95f2-8032dfa332e1")
+    hardware = FactoryGirl.create(:hardware,
+                                  :disks => [disk])
+
+    vm = FactoryGirl.create(:vm_redhat, :ext_management_system => @ems,
+                            :uid_ems     => "4f6dd4c3-5241-494f-8afc-f1c67254bf77",
+                            :ems_cluster => @cluster,
+                            :ems_ref     => "/api/vms/4f6dd4c3-5241-494f-8afc-f1c67254bf77",
+                            :storage     => storage,
+                            :storages    => [storage],
+                            :hardware    => hardware)
+    vm.with_relationship_type("ems_metadata") { vm.parent = @rp }
+
+    VCR.use_cassette("#{described_class.name.underscore}_target_existing_vm") do
+      EmsRefresh.refresh(vm)
+    end
+
+    assert_table_counts
+    assert_vm(vm, storage)
+    assert_vm_rels(vm, hardware, storage)
+    assert_cluster
+    assert_storage(storage, vm)
+  end
+
+  def assert_table_counts
+    expect(ExtManagementSystem.count).to eq(1)
+    expect(EmsCluster.count).to eq(1)
+    expect(ResourcePool.count).to eq(1)
+    expect(Vm.count).to eq(1)
+    expect(Storage.count).to eq(1)
+    expect(Disk.count).to eq(1)
+    expect(GuestDevice.count).to eq(1)
+    expect(Hardware.count).to eq(1)
+    expect(OperatingSystem.count).to eq(1)
+    expect(Snapshot.count).to eq(1)
+
+    expect(Relationship.count).to eq(6)
+    expect(MiqQueue.count).to eq(3)
+  end
+
+  def assert_vm(vm, storage)
+    vm.reload
+    expect(vm).to have_attributes(
+      :template               => false,
+      :ems_ref                => "/api/vms/4f6dd4c3-5241-494f-8afc-f1c67254bf77",
+      :ems_ref_obj            => "/api/vms/4f6dd4c3-5241-494f-8afc-f1c67254bf77",
+      :uid_ems                => "4f6dd4c3-5241-494f-8afc-f1c67254bf77",
+      :vendor                 => "redhat",
+      :raw_power_state        => "down",
+      :power_state            => "off",
+      :connection_state       => "connected",
+      :name                   => "123",
+      :format                 => nil,
+      :version                => nil,
+      :description            => nil,
+      :location               => "4f6dd4c3-5241-494f-8afc-f1c67254bf77.ovf",
+      :config_xml             => nil,
+      :autostart              => nil,
+      :host_id                => nil,
+      :last_sync_on           => nil,
+      :storage_id             => storage.id,
+      :last_scan_on           => nil,
+      :last_scan_attempt_on   => nil,
+      :retires_on             => nil,
+      :retired                => nil,
+      :boot_time              => nil,
+      :tools_status           => nil,
+      :standby_action         => nil,
+      :previous_state         => "up",
+      :last_perf_capture_on   => nil,
+      :registered             => nil,
+      :busy                   => nil,
+      :smart                  => nil,
+      :memory_reserve         => 1024,
```

Comment:
Although rubocop is asking for `_` every three digits for readability, it's preferable to use the ActiveSupport helper methods.

e.g. this line would read better as:

```ruby
:memory_reserve         => 1.kilobyte,
```